nightfall-optimist/src/services/event-queue.mjs

```diff
@@ -17,21 +17,23 @@ Web3js handles event removal (due to a chain reorg) by re-emitting the (now remo
 event with its `removed` property set to true. In the code below, we look out for
 and catch these removals, processing them appropriately.
 */
+/* eslint-disable import/no-cycle */
```

Comment:
Cyclic imports are an anti-pattern and so we should not just disable the check if one exists.

The specific block-assemble logic should be kept out of this file. There is a strong chance this event-queue code will need to re-used in `client` and thus elevated to a `common-files` level.

Instead a generic enqueue function would be preferred - this should also prevent the cyclic dependency